compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -449,6 +482,77 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
+                                                            != rcvr_ty
+                                                } else {
+                                                    false
+                                                }
+                                            }
+                                            (Mode::Path, false) => true,
+                                            _ => false,
+                                        }
+                                    } else {
+                                        false
+                                    }
+                                })
+                                .collect::<Vec<_>>();
+                            if inherent_impls_candidate.len() > 0 {
+                                inherent_impls_candidate.sort();
+                                inherent_impls_candidate.dedup();
+                                // number of type to shows at most.
+                                const LIMIT: usize = 3;
+                                let mut note = format!("The {item_kind} was found for");
+                                if inherent_impls_candidate.len() > 1 {
+                                    for impl_item in inherent_impls_candidate.iter().take(LIMIT - 2)
+                                    {
+                                        let impl_ty = self.tcx.at(span).type_of(*impl_item);
+                                        note = format!("{} {},", note, impl_ty);
+                                    }
+                                    let impl_ty = self.tcx.at(span).type_of(
+                                        inherent_impls_candidate
+                                            [inherent_impls_candidate.len() - 1],
+                                    );
+                                    if inherent_impls_candidate.len() > LIMIT {
+                                        note = format!("{} {},", note, impl_ty);
+                                    } else {
+                                        note = format!("{} {} and", note, impl_ty);
+                                    }
+                                }
+                                let impl_ty = self
+                                    .tcx
+                                    .at(span)
+                                    .type_of(*inherent_impls_candidate.last().unwrap());
+                                note = format!("{} {}", note, impl_ty);
+                                if inherent_impls_candidate.len() > LIMIT {
+                                    note = format!(
+                                        "{} and {} more",
+                                        note,
+                                        inherent_impls_candidate.len() - LIMIT
+                                    );
+                                }
+                                err.note(&format!("{}.", note));
```

Comment:
We don't capitalize or have a final period on sentences. We only break that convention if we have long text with multiple sentences, but it is heavily frown upon.